modules/debug-nodes-hcp.adoc

```diff
@@ -0,0 +1,111 @@
+----
++
+If you did not find any error messages in the status of the `HostedCluster` and `NodePool` resources, then follow the next steps.
+
+. Check if your worker machines are created by running the following commands, replacing values as necessary:
++
+[source,terminal]
+----
+$ HC_NAMESPACE="clusters"
+$ HC_NAME="cluster_name"
+$ CONTROL_PLANE_NAMESPACE="${HC_NAMESPACE}-${HC_NAME}"
+$ oc get machines.cluster.x-k8s.io -n $CONTROL_PLANE_NAMESPACE
+$ oc get awsmachines -n $CONTROL_PLANE_NAMESPACE
+----
+
+. If worker machines do not exist, check if the `machinedeployment` and `machineset` resources are created by running the following commands:
++
+[source,terminal]
+----
+$ oc get machinedeployment -n $CONTROL_PLANE_NAMESPACE
+$ oc get machineset -n $CONTROL_PLANE_NAMESPACE
+----
+
+. If the `machinedeployment` and `machineset` resources do not exist, check logs of the HyperShift Operator by running the following command:
++
+[source,terminal]
+----
+$ oc logs deployment/operator -n hypershift
+----
+
+. If worker machines exist but are not provisioned in the hosted cluster, check the log of the cluster API provider by running the following command:
++
+[source,terminal]
+----
+$ oc logs deployment/capi-provider -c manager -n $CONTROL_PLANE_NAMESPACE
+----
+
+. If worker machines exist and are provisioned in the cluster, ensure that machines are initialized through Ignition successfully by checking the system console logs. Check the system console logs of every machine by using the `console-logs` utility by running the following command:
++
+[source,terminal]
+----
+$ ./bin/hypershift console-logs aws --name $HC_NAME --aws-creds ~/.aws/credentials --output-dir /tmp/console-logs
+----
++
+You can access the system console logs in the `/tmp/console-logs` directory. The control plane exposes the Ignition endpoint. If you see the error related to the Ignition endpoint, then the Ignition endpoint is not accessible from the worker nodes through `https`.
+
+. If worker machines are provisioned and initialized through Ignition successfully, you can extract and access the journal logs of every worker machine by creating a bastion machine. A bastion machine allows you to access worker machines by using SSH.
+
+.. Create a bastion machine by running the following command:
++
+[source,terminal]
+----
+$ ./bin/hypershift create bastion aws --aws-creds ~/.aws/credentials --name $CLUSTER_NAME --ssh-key-file /tmp/ssh/id_rsa.pub
+----
+
+.. Extract the public and private key for the cluster by running the following commands:
++
+[souce,terminal]
+----
+$ mkdir /tmp/ssh
+$ oc get secret -n clusters ${HC_NAME}-ssh-key -o jsonpath='{ .data.id_rsa }' | base64 -d > /tmp/ssh/id_rsa
+$ oc get secret -n clusters ${HC_NAME}-ssh-key -o jsonpath='{ .data.id_rsa\.pub }' | base64 -d > /tmp/ssh/id_rsa.pub
+----
++
+[NOTE]
+====
+You can skip this step if you did not use the `--generate-ssh` flag when creating the cluster.
```

Comment:
What do you think of making this an optional step instead of a note? Something like `Optional: If you used the `--generate-ssh` flag when creating the cluster you can extract the public and private key for the cluster by running the following commands:`